it/uk/gov/hmrc/minorentityidentificationfrontend/controllers/AbstractBusinessVerificationControllerISpec.scala

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.minorentityidentificationfrontend.controllers
+
+import org.scalatest.BeforeAndAfterEach
+import play.api.test.Helpers._
+import uk.gov.hmrc.minorentityidentificationfrontend.assets.TestConstants._
+import uk.gov.hmrc.minorentityidentificationfrontend.featureswitch.core.config.{BusinessVerificationStub, FeatureSwitching}
+import uk.gov.hmrc.minorentityidentificationfrontend.models.KnownFactsMatchingResult.SuccessfulMatchKey
+import uk.gov.hmrc.minorentityidentificationfrontend.models.{BusinessVerificationPass, JourneyConfig, Registered}
+import uk.gov.hmrc.minorentityidentificationfrontend.stubs._
+import uk.gov.hmrc.minorentityidentificationfrontend.utils.WiremockHelper.{stubAudit, verifyAudit}
+import uk.gov.hmrc.minorentityidentificationfrontend.utils.{AuditEnabledSpecHelper, WiremockHelper}
+
+trait AbstractBusinessVerificationControllerISpec
+  extends AuditEnabledSpecHelper
+  with FeatureSwitching
+  with AuthStub
+  with BusinessVerificationStub
+  with StorageStub
+  with RegisterStub
+  with BeforeAndAfterEach
+  with WiremockHelper {
+
+  val businessVerificationResultUrlPrefix: String
+
+  val businessEntityBuilder: () => JourneyConfig
+
+  def commonTest(): Unit = {
```

Comment:
You could add an abstract value for the Utr which would enable you to use the appropriate Utr for Trusts and Unincorporated associations.

Reply:
Richard I am going to add one also for :
<img width="754" alt="Screenshot 2022-03-15 at 12 27 46" src="https://user-images.githubusercontent.com/4958044/158377619-22f13e5e-fbc7-418c-8c2f-dac156e38805.png">
because it will used for the story Refaque is playing https://github.com/hmrc/minor-entity-identification-frontend/tree/SAR-9324